sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -980,21 +985,40 @@ def download_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> Do
         return DownloadManifestResult(digest=digest, data=manifest_stream, manifest=manifest)
 
     @distributed_trace
-    def download_blob(self, repository: str, digest: str, **kwargs) -> DownloadBlobResult:
-        """Download a blob that is part of an artifact.
+    def download_blob(self, repository: str, digest: str, **kwargs) -> DownloadBlobStream:
+        """Download a blob that is part of an artifact to a stream.
 
-        :param str repository: Name of the repository
+        :param str repository: Name of the repository.
         :param str digest: The digest of the blob to download.
-        :returns: DownloadBlobResult
-        :rtype: ~azure.containerregistry.DownloadBlobResult
+        :returns: DownloadBlobStream
+        :rtype: ~azure.containerregistry.DownloadBlobStream
+        :raises ValueError: If the requested digest does not match the digest of the received blob.
         """
-        deserialized = self._client.container_registry_blob.get_blob(repository, digest, **kwargs)
-
-        blob_content = b''
-        for chunk in deserialized: # type: ignore
-            if chunk:
-                blob_content += chunk
-        return DownloadBlobResult(data=BytesIO(blob_content), digest=digest)
+        chunk_size = DEFAULT_CHUNK_SIZE
+        first_chunk, headers = cast(
+            Tuple[PipelineResponse, Dict[str, str]],
+            self._client.container_registry_blob.get_chunk(
+                repository,
+                digest,
+                range_header=f"bytes=0-{chunk_size}",
+                cls=_return_response_and_headers,
+                **kwargs
+            )
+        )
+        return DownloadBlobStream(
+            response=first_chunk,
+            digest=digest,
+            get_next=functools.partial(
+                self._client.container_registry_blob.get_chunk,
+                name=repository,
+                digest=digest,
+                cls=_return_response_and_headers,
+                **kwargs
+            ),
+            blob_size=int(headers["Content-Range"].split("/")[1]),
+            downloaded=int(headers["Content-Length"]),
+            chunk_size=chunk_size
+        )
```

Comment:
Double checked the `None` value validation here and in async: it's not necessary to do the validation because the value info `ValueError: No value for given attribute` is pretty clear.